tests/modeladmin/tests.py

```diff
@@ -619,6 +620,63 @@ def test_log_actions(self):
         self.assertEqual(ma.log_change(mock_request, self.band, 'changed'), LogEntry.objects.latest('id'))
         self.assertEqual(ma.log_change(mock_request, self.band, 'deleted'), LogEntry.objects.latest('id'))
 
+    def test_get_deleted_objects_called(self):
+        """
+        Test that the new ModelAdmin get_deleted_objects() method returns expected results from
+        django.contrib.admin.utils.get_deleted_objects.
+        """
+        mock_request = MockRequest()
+        mock_request.POST = {}
+        mock_request.META = {'SCRIPT_NAME': 'test'}
+        mock_request.user = User.objects.create_superuser(username='bob', email='bob@test.com', password='test')
+
+        ma = ModelAdmin(Band, self.site)
+
+        deletable_objects, model_count, perms_needed, protected = ma.get_deleted_objects(
+            [self.band], ma.opts, request, DEFAULT_DB_ALIAS
+        )
+        self.assertEqual(deletable_objects, ['Band: The Doors'])
+        self.assertEqual(model_count, {'bands': 1})
+        self.assertEqual(perms_needed, set())
+        self.assertEqual(protected, [])
+
+    def test_overriding_get_deleted_objects(self):
+        """
+        Test that delete_selected action uses the modeladmin's custom get_deleted_objects method
+        """
+        mock_request = MockRequest()
+        mock_request.POST = {}
+        mock_request.META = {'SCRIPT_NAME': 'test'}
+        mock_request.user = User.objects.create_superuser(username='bob', email='bob@test.com', password='test')
+
+        class BandAdmin(ModelAdmin):
+            def get_deleted_objects(self, objs, opts, request, using):
+                return ['a deletable object'], {'bands': 1}, set(), []
+
+        ba = BandAdmin(Band, self.site)
+        resp = delete_selected(ba, mock_request, Band.objects.filter(id=self.band.id))
+
+        self.assertEqual(resp.context_data['deletable_objects'], [['a deletable object']])
+        self.assertEqual(resp.context_data['model_count'], {'bands': 1}.items())
+
+    def test_default_get_deleted_objects(self):
+        """
+        Test that delete_selected action uses the default get_deleted_objects method if no custom method defined
+        """
+        mock_request = MockRequest()
+        mock_request.POST = {}
+        mock_request.META = {'SCRIPT_NAME': 'test'}
+        mock_request.user = User.objects.create_superuser(username='bob',
```

Comment:
Use hanging indent as described in [Python coding style](https://docs.djangoproject.com/en/dev/internals/contributing/writing-code/coding-style/#python-style).